crates/gfx-plugin/src/lib.rs

```diff
@@ -74,21 +80,86 @@ where
         System: IntoSystem<Parameters>,
         Parameters: SystemParameters,
     {
-        self.builder = self.builder.add_systems(systems);
+        self.app_builder = self.app_builder.add_systems(systems);
         self
     }
 
     fn build(self) -> Self::App {
-        let (graphics_engine, graphics_engine_handle) = GraphicsEngine::new()
+        let mut graphics_options_builder = GraphicsOptionsBuilder::default();
+        if let Some(speed) = self.camera_movement_speed {
+            graphics_options_builder.camera_movement_speed(speed);
+        }
```

Comment:
Is there any reason for why not storing the builder rather than the parameters and allowing the "setters" to change the `graphics_option_builder`? 
Feel like there would be less code over all. The only code you would have here is just the `build` call in 
```rust  
let renderer_options = renderer_options_builder
            .build()
            .expect("default renderer options should be valid");```

Reply:
Ah, that's smart. No reason, I just threw it together extremely quickly. I'll change it to that!